### ISSUE TITLE
Stems and branches of month and day in Chinese calendar

### ISSUE THREAD

**Issue by dengyiming1983** [MenoData/Time4J#910]
I want to get the Stem and Branch of months and days.
  
PlainDate plainDate = PlainDate.of(2020, 7, 14);
        ChineseCalendar calendar = plainDate.transform(ChineseCalendar.class);
        System.out.println(calendar.transform(PlainDate.axis()));
        System.out.println(calendar.getYear().getStem().getDisplayName(Locale.CHINESE) +
                calendar.getYear().getBranch().getDisplayName(Locale.CHINESE) + "年");
        System.out.println(calendar.getMonth().getDisplayName(Locale.CHINESE, NumberSystem.JAPANESE));
        System.out.println(calendar.getDayOfMonth());
        // output
//        2020-07-14
//        庚子年 五月 24

 System.out.println(calendar.getMonth().getStem().getDisplayName(Locale.CHINESE) +
                calendar.getMonth().getBranch().getDisplayName(Locale.CHINESE) + "月");
// 癸未月 

System.out.println(calendar.getDayOfMonth().getStem().getDisplayName(Locale.CHINESE) +
                calendar.getDayOfMonth().getBranch().getDisplayName(Locale.CHINESE) + "日");
/ /戊午日

**Comment by MenoData** [comment]
Your code will not compile because month and day-of-month are not cyclic by calendar design, in contrast to Chinese cyclic years which indeed use a sixty year cycle consisting of stems and branches. Months and day-of-month don't know such a cyclic design but are just integers with the special option of leap months (using a leap indicator: in Chinese it is the sign 閏).

However, I believe, you just want to look for printing month and day-of-month in Chinese language. Here an example using the class `ChronoFormatter` (which is also capable of print leap indicators):

```
        PlainDate plainDate = PlainDate.of(2020, 7, 14);
        ChineseCalendar calendar = plainDate.transform(ChineseCalendar.class);

        System.out.println(calendar.getYear().getStem().getDisplayName(Locale.CHINESE) +
            calendar.getYear().getBranch().getDisplayName(Locale.CHINESE) + "年");
        // output of cyclic year is the same as produced in formatters by pattern symbol U: 庚子年

        ChronoFormatter<ChineseCalendar> f1 = // M月 produces integer form: 5月
            ChronoFormatter
                .ofPattern("U年 M月 d日", PatternType.CLDR, Locale.CHINESE, ChineseCalendar.axis());

        ChronoFormatter<ChineseCalendar> f2 = // MMMM produces text form: 五月
            ChronoFormatter
                .ofPattern("U年 MMMM d日", PatternType.CLDR, Locale.CHINESE, ChineseCalendar.axis())
                .with(Attributes.NUMBER_SYSTEM, NumberSystem.JAPANESE);

        System.out.println(f1.print(calendar)); // 庚子年 5月 24日
        System.out.println(f2.print(calendar)); // 庚子年 五月 二十四日
```

Finally a question to you: Can you confirm that Chinese language uses the same number signs as Japanese? I am not an expert for Chinese language so awaiting your answer. The constant `NumberSystem.JAPANESE` was mainly intended for the Japanese calendar. If Chinese language uses the same characters then I could copy the Japanese constant to a new constant `NumberSystem.CHINESE`.

**Comment by dengyiming1983** [comment]
Thank you very much for your prompt reply.

I can't find "numbersystem.Chinese" so I'm using "numbersystem.japanese". 
I'm not sure about the Japanese calendar.  but chinese have  "month" and" dayOfMonth" of stems and branches.
![image](https://user-images.githubusercontent.com/13852039/87497494-3b3e1e00-c688-11ea-8c47-1439205bef9e.png)
"2020-07-14"  equal "庚子年  癸未月  戊午日"
"庚子年  癸未月  戊午日" are very important in China.

Can you add the month  of stems and branches  and day of month  of stems and branches?

**Comment by MenoData** [comment]
Okay, now I think to have understood you. Sorry that I have not yet implemented this desired feature.  After some research, it seems that I should be able to implement it on the base of the book "Calendrical Calculatiions (ulitimate edition)" by Dershowitz/Reingold (page 320). The necessary refactoring and enhancement api design is still for debate so I will need some time (and soon my summer holidays start ;-)). About Chinese number signs, I will have a deeper look, too.

**Comment by dengyiming1983** [comment]
> Okay, now I think to have understood you. Sorry that I have not yet implemented this desired feature. After some research, it seems that I should be able to implement it on the base of the book "Calendrical Calculatiions (ulitimate edition)" by Dershowitz/Reingold (page 320). The necessary refactoring and enhancement of api design is still for debate because stems and branches are actually only related to cyclic years so I will need some time (and soon my summer holidays start ;-)). About Chinese number signs, I will have a deeper look, too.

Chinese months are calculated according to the 24 solar terms of the calendar
"LICHUN" is the first month of every year. month branch is "寅" //  X寅月 XX日

chinese month ( stem branches month  like X寅月)
solar terms    month      branches (time4j number of braches )
LICHUN           1             寅 (2)
JINGZHE          2             卯 (3)
QINGMING      3             辰 (4)
LIXIA 	        4             巳 (5)
MANGZHONG   5           午 (6)
XIAOSHU          6             未 (7)
LIQIU                7             申 (8)
BAILU               8             酉 (9)
HANLU             9             戌 (10)
LIDONG           10            亥 (11)
DAXUE             11            子 (0)
XIAOHAN         12            丑 (1)

Haven't figured stem month out how to do that yet. 
I think it's also based on some day to start. I'm going to take stem as a loop. 
Calculate the number of months of solar terms   % 10 as stem . end month base on “LICHUN” number of stem. 
Number of branches  =  ( number of months of solar terms of the current year + 2 ) % 12

chinese day
The largest common multiple of 10 and 12.    60 a cycle, like the cycle year. Use a certain day as a benchmark。
example : 
"1900-02-20 " chinese day is "甲子日" ， "1900-04-21" is  "甲子日" after 60 days.  
There is a 60 day difference between the start date and the end date.
 First calculate the number of days of difference.   
stem branches day like cycle year.

**Comment by MenoData** [comment]
I have now implemented sexagesimal names of both days and months (in latter case rather based on solar terms instead of lunar months). About the latter one: The book of Dershowitz/Reingold was of no help so I had to develop my own algorithm in a heuristic manner (causing some headache ;-)) but did extensive and successful tests. The last commit contains the tests so you might want to inspect it, too.

Your input was great and very valuable so thank you very much.
